bound/bound.go

```diff
@@ -11,22 +11,32 @@ type Bound struct {
 	Min, Max float64
 }
 
+// IsEmpty returns whether the bound is empty
+func (b *Bound) IsEmpty() bool {
+	return math.IsNaN(b.Min) || math.IsNaN(b.Max)
+}
+
+// IsValid returns whether the bound is valid
+func (b *Bound) IsValid() bool {
+	return !math.IsNaN(b.Min) && !math.IsNaN(b.Max) && b.Min <= b.Max
```

Comment:
You don't need the NaN checks here since any equality test except `!=` always returns false if there is at least one NaN.

Reply:
Should this be a package-level function?
What about intervals with inverted min max?

Reply:
Ah, sorry, I'm on my phone and I didn't see the end of the line.

Reply:
If it stays a method, it doesn't need a pointer receiver